packages/admin/cms-admin/src/generator/generateForm.ts

```diff
@@ -0,0 +1,358 @@
+import { IntrospectionEnumType, IntrospectionField, IntrospectionNamedTypeRef, IntrospectionObjectType, IntrospectionQuery } from "graphql";
+
+import { CrudGeneratorConfig } from "./types";
+import { buildNameVariants } from "./utils/buildNameVariants";
+import { camelCaseToHumanReadable } from "./utils/camelCaseToHumanReadable";
+import { findRootBlocks } from "./utils/findRootBlocks";
+import { writeGenerated } from "./utils/writeGenerated";
+
+export async function writeCrudForm(generatorConfig: CrudGeneratorConfig, schema: IntrospectionQuery): Promise<void> {
+    const { target: targetDirectory, entityName } = generatorConfig;
+    const { classNamePlural, classNameSingular, instanceNamePlural } = buildNameVariants(entityName);
+    const rootBlocks = findRootBlocks(generatorConfig, schema);
+    const instanceEntityName = entityName[0].toLowerCase() + entityName.substring(1);
+
+    const schemaEntity = schema.__schema.types.find((type) => type.kind === "OBJECT" && type.name === entityName) as
+        | IntrospectionObjectType
+        | undefined;
+    if (!schemaEntity) throw new Error("didn't find entity in schema types");
+    const formFields = schemaEntity.fields
+        .filter((field) => {
+            if (field.name === "id" || field.name === "updatedAt" || field.name === "createdAt" || field.name === "scope") return false;
+            return true;
+        })
+        .filter((field) => {
+            let type = field.type;
+            if (type.kind == "NON_NULL") type = type.ofType;
+            if (type.kind == "LIST") return false;
+            if (type.kind == "OBJECT") return false; //TODO support nested objects
+            return true;
+        });
```

Comment:
The formFields should probably be derived from the InputType, not the ObjectType. 

For example, I explicitly told the API generator to not add `totalAmountPaid` to the input in the following entity:

```ts
@Entity()
@RootBlockEntity()
export class Order extends BaseEntity<Order, "id"> implements DocumentInterface {
    // ...

    @Property({ type: DecimalType })
    @Field(() => Float)
    @CrudField({
        input: false,
    })
    totalAmountPaid: number;

     // ...
}
```

Yet, the admin generator generates a form field for this field:

```tsx
// ...

type FormValues = Omit<GQLOrderFormFragment, "totalAmountPaid"> & {
    totalAmountPaid: string;
};

// ...

export function OrderForm({ id }: FormProps): React.ReactElement {
    
    // ...

    return (
        // ...
          <Field
              required
              fullWidth
              name="totalAmountPaid"
              component={FinalFormInput}
              type="number"
              label={<FormattedMessage id="order.totalAmountPaid" defaultMessage="Total Amount Paid" />}
          />
        // ...
    );
}
```

Reply:
fixed